keras_nlp/samplers/sampler_test.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Sampler classes."""
+
+import tensorflow as tf
+
+import keras_nlp
+from keras_nlp.samplers.greedy import Greedy
+
+
+class SamplerTest(tf.test.TestCase):
+    def test_serialization(self):
+        sampler = keras_nlp.samplers.Greedy()
+        config = keras_nlp.samplers.serialize(sampler)
+        expected_config = {
+            "class_name": "Greedy",
+            "config": {
+                "jit_compile": True,
+            },
+        }
+        self.assertDictEqual(expected_config, config)
+
+    def test_deserialization(self):
+        # Test get from string.
+        identifier = "greedy"
+        sampler = keras_nlp.samplers.get(identifier)
+        self.assertIsInstance(sampler, Greedy)
+
+        # Test string is not case-sensitive.
+        identifier = "Greedy"
```

Comment:
Do we need this? I was suggesting to remove above. Given that we are a new library with no legacy constraints (unlike core Keras), I don't think we need to support all these case variants.

We are in the position to make our code more uniform! Either a "snake_case_identifier" or an actual  `Greedy()` object.